ecl/hql/hqlgram2.cpp

```diff
@@ -11006,14 +11201,24 @@ void HqlGram::syntaxError(const char *s, int token, int *expected)
     {
         msg.append("Unknown identifier");
         if (yytext && *yytext)
+        {
             msg.append(" \"").append(yytext).append('\"');
+            if(lookupCtx.queryParseContext().autoIdSuggestions)
```

Comment:
This logic probably needs improving.  You should only report a suggestion of an alternative identifier if a valid predefined identifier is valid at this point.

If no identifier is valid (and if it is) you should be auto-suggesting against the list of keywords that are valid next.  Leave that to a secondary PR though - it would be better to get this one merged first.